network/peers/connections/conn_gater.go

```diff
@@ -89,11 +91,11 @@ func (n *connGater) InterceptAccept(multiaddrs libp2pnetwork.ConnMultiaddrs) boo
 // InterceptSecured is called for both inbound and outbound connections,
 // after a security handshake has taken place and we've authenticated the peer.
 func (n *connGater) InterceptSecured(direction libp2pnetwork.Direction, id peer.ID, multiaddrs libp2pnetwork.ConnMultiaddrs) bool {
-	if n.isBadPeer(n.logger, id) {
-		n.logger.Debug("rejecting inbound connection due to bad peer", fields.PeerID(id))
-		return false
+	if direction == libp2pnetwork.DirOutbound {
+		return n.canConnectOutbound(n.logger, id)
+	} else {
+		return n.canConnectInbound(n.logger, id)
 	}
```

Comment:
Probably want to also handle `DirUnknown` here ? If that's a thing

Maybe explicitly check for it - to show we are aware it's a possibility